MaxText/pyconfig.py

```diff
@@ -392,16 +409,15 @@ def configure_gpt3_task(raw_keys):
     """dynamically configure gpt3 task based on training rules"""
     # follow https://github.com/google/paxml/blob/19db52eed85ae0d2365339b83a97cd0b873bbf73/paxml/tasks/lm/params/c4.py#L280
     #   according to training_rules of mlperf gpt3 training
-    global_batch_size = calculate_global_batch_sizes(raw_keys)[1]
-    if global_batch_size <= 3584:
+    global_batch_size_to_train_on = raw_keys["global_batch_size_to_train_on"]
+    if global_batch_size_to_train_on <= 3584:
       raw_keys["learning_rate"] = 2e-5
     else:
       raw_keys["learning_rate"] = 3e-5
-    warmup_steps = math.ceil(265.0 * 1536 / global_batch_size - 1e-6)
-    decay_end_step = math.ceil(108600.0 * 1536 / global_batch_size - 1e-6)
+    warmup_steps = math.ceil(265.0 * 1536 / global_batch_size_to_train_on - 1e-6)
+    decay_end_step = math.ceil(108600.0 * 1536 / global_batch_size_to_train_on - 1e-6)
     raw_keys["learning_rate_schedule_steps"] = decay_end_step
     raw_keys["warmup_steps_fraction"] = warmup_steps / decay_end_step
-    global_batch_size_to_train_on = calculate_global_batch_sizes(raw_keys)[1]
     if raw_keys["dataset_type"] != "synthetic":
       raw_keys["eval_interval"] = math.ceil(24567 / global_batch_size_to_train_on)
```

Comment:
Hi, @aireenmei 

Previous in gpt3 mlperf submission, I set `eval_interval=-1` and let this line update accordingly. 
 In the following changes above, we actually need a positive raw `eval_interval`
```
    if raw_keys["eval_interval"] > 0:
      if raw_keys["eval_per_device_batch_size"] <= 0:
        raw_keys["eval_per_device_batch_size"] = raw_keys["per_device_batch_size"]

      (
          raw_keys["global_batch_size_to_load_eval"],
          raw_keys["global_batch_size_to_eval_on"],
          raw_keys["micro_batch_size_to_eval_on"],
      ) = calculate_global_batch_sizes(
          raw_keys["eval_per_device_batch_size"], raw_keys["expansion_factor_real_data"], get_num_target_devices(raw_keys), 1
      )
```
I do expect there might be some conflicts. Let me know if it would be easier to change in this PR or shall we just change `eval_interval` to some positive number by default accordingly?

Reply:
The current behavior (dataset_type in (tfds, hf, grain)) is that eval will only run when `eval_interval>0` (https://github.com/AI-Hypercomputer/maxtext/blob/5d1e7a37d03573edd2b006330158605f8a918d57/MaxText/train.py#L685), and only in this case we need these eval related hyperparams. When dataset_type=c4_mlperf, it initiates eval iterator regardless of "eval_interval" and has it's own fundtion `get_eval_global_batch_size_to_load` https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/input_pipeline/_tfds_data_processing_c4_mlperf.py#L350. I think this change should not cause issue for c4_mlperf. But we can probably align on the use of `eval_interval` as the switch for eval? Wdyt?

Reply:
Thank you @aireenmei for explanation!